src/foam/nanos/crunch/AuthenticatedCapabilityDAO.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+foam.CLASS({
+  package: 'foam.nanos.crunch',
+  name: 'AuthenticatedCapabilityDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  javaImports: [
+    'foam.nanos.auth.*'
+  ],
+
+  documentation: `
+  Only admin users have write access to capabilityDAO, non-admin users have read access depending on their permissions.
+  `,
+
+  methods: [
+    {
+      name: 'checkPermission',
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        }
+      ],
+      documentation: `Checks if a user is admin`,
+      javaCode: `
+      User user = (User) x.get("user");
+      if ( user == null ) throw new AuthorizationException();
+      AuthService auth = (AuthService) x.get("auth");
+      boolean hasPermission = auth.check(x, "service.*");
```

Comment:
why do you check for `service.*`? Shouldn't it be something like `capabilityDAO.read`, `...create` etc?
Also the class should be named AuthorizedCapabilityDAO since it's checking for permissions

Reply:
this can be ignored though as I think we will replace this whole thing with your StandardAuthorizer

Reply:
ya true, I can just get rid of this easy fix haha